api/v1alpha1/gateway_types.go

```diff
@@ -70,10 +70,24 @@ const (
 
 // Listener defines a
 type Listener struct {
-	// Name can be used to tie this Listener to a ListenerStatus entry with the
-	// same name. Each listener must have a unique name within a Gateway. This
-	// must be a valid DNS_LABEL.
-	Name string `json:"string"`
+	// Name is the listener's name and should be specified as an
+	// RFC 1035 DNS_LABEL [1]:
+	//
+	// [1] https://tools.ietf.org/html/rfc1035
+	//
+	// Each listener of a Gateway must have a unique name. Name is used
+	// for the following features:
+	//
+	// * Associating a listener in ListenerStatus.
+	//
+	// * Generating a wildcard certificate for the subdomain of Name when
+	//   the TLS configuration of an HTTPS listener does not include a
+	//   certificate.
```

Comment:
I don't see any mention of wildcard certificates elsewhere in the spec. If this is a new feature, we should first capture a use case and then propose it independently.